### DISCUSSION TITLE
[Question] Dramatic increase in file size after editing

### DISCUSSION THREAD

**Discussion by Chris2137** [hm21/pro_image_editor#93] [Q&A] [answered]
### Platforms

iOS

### Your question

Hi developers,
  Thanks for developing this fantastic plugin, it is nice and good to use. Just one thing that I would like to confirm if that's expected behavior. 
  I downloaded the source and try out with the example, just adding minimal code to print the image size before and after edit, and found the size increase dramatically (from 321KB to 2.9M). The editing is minimal too, just drawing a circle.
![image](https://github.com/hm21/pro_image_editor/assets/22078631/2ba7c06a-9bec-445f-9d49-e4dbdfffaa8e)

Thanks again for looking into this. Cheers.

**Comment by hm21** [comment, marked as answer]
Hi Chris,

Yes, this “dramatic” increase is normal. When you press the “Done” button, it will generate a PNG where it adds every pixel calculated from the width and height of the image. This gives the user the highest possible quality, even when adding emojis, stickers and other stuff. 
This “dramatic” increase you can see, especially good when you insert an optimized image. The image you post above comes from [picsum](https://picsum.photos) which provides random images for development. For fast and small transform size use “picsum” [fastly](https://www.fastly.com/products/image-optimization) which can change the format to like JPEG, AVIF, WebP and also optimize your image directly. So in your case, you are comparing an “original” image with an optimized “image”. For example, if you try the same with the sample image from the asset folder, the “increase” will be much smaller.

Anyway, in most cases, when the developer creates a high-end application, he will upload the generated image directly to his server where it will convert the image to new formats with optimization. The other option is to use a tool like “fastly” which will convert the original automatically but still require uploading the “original” generated image.
For projects which don't have a server backend or a slow backend, it is possible to convert the image locally. I recommend you in this case to convert the image to “JPEG” because WebP and AVIF are not supported on all mobile devices. In the dev branch can you checkout [this example](https://github.com/hm21/pro_image_editor/blob/94dc5b11eb7a430772e4c224bb0d0fb90881cc2e/example/lib/pages/image_format_convert_example.dart) how you can convert the image.

**Comment by Chris2137** [comment]
Thanks for your response @hm21 , I tried to compress the way same as the example, the image increases to around double of original size, does that sound normal to you?

**Comment by hm21** [comment]
Yes, that's normal that the original is the double size compared to a full optimized image. In the example, it just changes the format to “JPEG”. For a high-quality optimization, should you use a server backend. You can try this [tool](https://www.onlineimagetool.com/en/convert-png-jpg-webp-gif) where you paste your image. It will show you directly how much it can optimize your image size when you insert the original image.
